### PR TITLE
Resampler.__iter__() not working

### LINKED ISSUE #15314: Resampler.__iter__() not working
#### Code Sample, a copy-pastable example if possible

```python
for i, k in df.resample('W-MON'):
    print(i, k)
```
```sh
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
<ipython-input-60-5ca9218261db> in <module>()
----> 1 for i, k in df.resample('W-MON'):
      2     print(i, k)
      3     break

/xxx/lib/python3.5/site-packages/pandas/core/groupby.py in __iter__(self)
    629         for each group
    630         """
--> 631         return self.grouper.get_iterator(self.obj, axis=self.axis)
    632
    633     @Substitution(name='groupby')

AttributeError: 'NoneType' object has no attribute 'get_iterator'
```
#### Problem description
can't iterate the resampler's key and grouped dataframe

#### Expected Output
iterate the resampler's key and the grouped dataframe, now workaround
```python
for key in df.resample('W-MON').groups.keys():
  try:
    print(key, df.resample('W-MON').get_group(key))
  except KeyError:
    continue
```

#### Output of ``pd.show_versions()``

<details>
INSTALLED VERSIONS
------------------
commit: None
python: 3.5.2.final.0
python-bits: 64
OS: Darwin
OS-release: 15.6.0
machine: x86_64
processor: i386
byteorder: little
LC_ALL: en_GB.UTF-8
LANG: en_GB.UTF-8
LOCALE: en_GB.UTF-8

pandas: 0.19.2
nose: None
pip: 9.0.1
setuptools: 30.2.0
Cython: 0.25.2
numpy: 1.12.0
scipy: 0.18.1
statsmodels: None
xarray: None
IPython: 5.1.0
sphinx: 1.5.1
patsy: 0.4.1
dateutil: 2.6.0
pytz: 2016.10
blosc: None
bottleneck: None
tables: None
numexpr: None
matplotlib: 1.5.3
openpyxl: 2.4.1
xlrd: None
xlwt: None
xlsxwriter: None
lxml: None
bs4: None
html5lib: 0.9999999
httplib2: None
apiclient: None
sqlalchemy: None
pymysql: None
psycopg2: 2.6.2 (dt dec pq3 ext lo64)
jinja2: 2.8
boto: None
pandas_datareader: None
</details>

----
update the workaround a bit

### PR DESCRIPTION
**Pull request by discort**
- [x] closes #15314
- [x] tests added / passed
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [x] whatsnew entry

### CODE REVIEW

File: doc/source/whatsnew/v0.24.0.txt [hunk 1]
```diff
@@ -182,6 +182,7 @@ Other Enhancements
 - :func:`to_timedelta` now supports iso-formated timedelta strings (:issue:`21877`)
 - :class:`Series` and :class:`DataFrame` now support :class:`Iterable` in constructor (:issue:`2193`)
 - :class:`DatetimeIndex` gained :attr:`DatetimeIndex.timetz` attribute. Returns local time with timezone information. (:issue:`21358`)
+- :class:`Resampler` now is iterable (:issue:`15314`).
```
Comment (TomAugspurger) [inline doc/source/whatsnew/v0.24.0.txt]: Add a note say what each item is, or make the analogy to group by. Could you also make a new subsection of the docs and link to it from here.

File: pandas/core/resample.py [hunk 2]
```diff
@@ -97,6 +97,18 @@ def __getattr__(self, attr):
 
         return object.__getattribute__(self, attr)
 
+    def __iter__(self):
+        """
+        Resampler iterator
```
Comment (TomAugspurger) [inline pandas/core/resample.py]: Maybe a bit more explanation here.

File: pandas/core/resample.py [hunk 3]
```diff
@@ -97,6 +97,18 @@ def __getattr__(self, attr):
...
+
+        Returns
+        -------
+        Generator yielding sequence of (name, subsetted object)
+        for each group
+        """
```
Comment (TomAugspurger) [inline pandas/core/resample.py]: Add an examples and a See Also to grouupby iter

File: doc/source/whatsnew/v0.24.0.txt (hunk 1 above)
Comment (discort) [reply doc/source/whatsnew/v0.24.0.txt]: is it looking good?
```
:class:`Resampler` now is iterable like :class:`GroupBy` (:issue:`15314`).
```
______

> Could you also make a new subsection of the docs and link to it from here.

I'm not sure what do you mean here. Could you please explain a little bit?

File: pandas/core/resample.py (hunk 2 above)
Comment (discort) [reply pandas/core/resample.py]: Do we have any example? I found only for `GroupBy`?
https://github.com/pandas-dev/pandas/blob/v0.23.4/pandas/core/groupby/groupby.py#L881-L890

**Comment by TomAugspurger** [comment]
dunder methods like __iter__ are included in the docs.

Although,
http://pandas-docs.github.io/pandas-docs-travis/generated/pandas.core.groupby.GroupBy.__iter__.html
isn't that great of a docstring at the moment :)

File: doc/source/whatsnew/v0.24.0.txt (hunk 1 above)
Comment (TomAugspurger) [reply doc/source/whatsnew/v0.24.0.txt]: A section like http://pandas-docs.github.io/pandas-docs-travis/groupby.html#iterating-through-groups, but in doc/source/timeseries.rst

**Review by jreback — CHANGES_REQUESTED** [review]

File: doc/source/timeseries.rst [hunk 4]
```diff
@@ -703,6 +703,23 @@ regularity will result in a ``DatetimeIndex``, although frequency is lost:
...
 
+.. _timeseries.iterating:
+
+Iterating through groups
+------------------------
+
+With the Resampler object in hand, iterating through the grouped data is very
+natural and functions similarly to :py:func:`itertools.groupby`:
+
+.. ipython::
```
Comment (jreback) [inline doc/source/timeseries.rst]: use ``..ipython:: python``

File: doc/source/timeseries.rst [hunk 5]
```diff
@@ -703,6 +703,23 @@ regularity will result in a ``DatetimeIndex``, although frequency is lost:
...
+
+   In [1]: resampled = df.resample('H')
```
Comment (jreback) [inline doc/source/timeseries.rst]: don't add the ipython prompts, they will be added during execution

File: doc/source/timeseries.rst [hunk 6]
```diff
@@ -703,6 +703,23 @@ regularity will result in a ``DatetimeIndex``, although frequency is lost:
 
    ts2[[0, 2, 6]].index
 
+.. _timeseries.iterating:
+
+Iterating through groups
+------------------------
+
+With the Resampler object in hand, iterating through the grouped data is very
```
Comment (jreback) [inline doc/source/timeseries.rst]: can you add a class ref here

File: doc/source/timeseries.rst [hunk 7]
```diff
@@ -703,6 +703,23 @@ regularity will result in a ``DatetimeIndex``, although frequency is lost:
 
    ts2[[0, 2, 6]].index
 
+.. _timeseries.iterating:
```
Comment (jreback) [inline doc/source/timeseries.rst]: can you add a reference to the groupby.iterating section (and a reverse ref from there to here)

File: doc/source/timeseries.rst (hunk 5 above)
Comment (jreback) [reply doc/source/timeseries.rst]: can you also fix the groupby.iterating example as well

**Review by jreback — CHANGES_REQUESTED** [review]
some doc comments. ping on green.

File: doc/source/timeseries.rst (hunk 7 above)
Comment (discort) [reply doc/source/timeseries.rst]: @jreback 

I didn't find how to add a reference in the [docs](https://pandas.pydata.org/pandas-docs/stable/contributing.html). Could you share how to do that?

**Comment by discort** [comment]
@jreback

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
thanks @discort

**Pull request merged**